src/adm/filter.rs

```diff
@@ -107,6 +115,104 @@ impl AdmFilter {
         l_sentry::report(&merged_tags, sentry::event_from_error(error));
     }
 
+    /// check to see if the bucket has been modified since the last time we updated.
+    pub async fn requires_update(&self) -> HandlerResult<bool> {
+        // don't update non-bucket versions (for now)
+        if !self.is_cloud() {
+            return Ok(false);
+        }
+        let bucket: url::Url = match self.source.parse() {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(HandlerError::internal(&format!(
+                    "Invalid bucket url: {:?} {:?}",
+                    self.source, e
+                )))
+            }
+        };
+        let host = match bucket.host() {
+            Some(v) => v,
+            None => {
+                return Err(HandlerError::internal(&format!(
+                    "Missing bucket Host {:?}",
+                    self.source
+                )))
+            }
+        }
+        .to_string();
+
+        let obj: cloud_storage::Object = cloud_storage::Object::read(&host, bucket.path())
+            .await
+            .map_err(|e| {
+                HandlerError::internal(&format!("Could not read bucket {:?}, {:?}", self.source, e))
+            })?;
+        if let Some(updated) = self.last_updated {
+            // if the bucket is older than when we last checked, do nothing.
+            return Ok(chrono::DateTime::<chrono::Utc>::from_utc(
+                chrono::NaiveDateTime::from_timestamp(
+                    updated
+                        .duration_since(std::time::UNIX_EPOCH)
+                        .unwrap_or_else(|_| std::time::Duration::from_secs(0))
+                        .as_secs() as i64,
+                    0,
+                ),
+                chrono::Utc,
+            ) <= obj.updated);
```

Comment:
Partly because I was hitting issues with the result being in DateTime and the UTC being in SystemTime, and just being frustrated about it all. Good call, I'll fix it.